state/runtime/evm/instructions.go

```diff
@@ -842,6 +843,14 @@ func opReturnDataCopy(c *state) {
 	}
 
 	data := c.returnData[dataOffset.Uint64():dataEndIndex]
+	// this should not happen because the memory is allocated considering both memOffset and data length and
+	// in the case of length==0 the fn returns immediately, not reaching this point
+	if uint64(len(c.memory)) < memOffset.Uint64()+uint64(len(data)) {
+		c.exit(errReturnDataOutOfBounds)
+
+		return
+	}
+
 	copy(c.memory[memOffset.Uint64():], data)
```

Comment:
Comparing with go ethereum, it seems like we should copy data limited only to this range:
`copy(m.store[offset:offset+size], value)`

Which in our case would be like this: 
`copy(c.memory[memOffset.Uint64():memOffset.Uint64()+length.Uint64()], data)`
Though it seems it is pretty similar to what is currently implemented, however not sure if we could rewrite some data accidentaly, because we haven't specified upper bound?